ext/tests/invalid_post_callback_signature.phpt

```diff
@@ -33,4 +31,5 @@ TestClass::test();
 --EXPECTF--
 string(3) "pre"
 string(4) "test"
-OpenTelemetry: post hook invalid signature, class=TestClass function=test
+
+Warning: TestClass::test(): OpenTelemetry: post hook invalid signature, class=TestClass function=test in %s on line %d
```

Comment:
with the improvement of using php_error_docref, it looks like the second part of the message is no longer required and could be removed now ("class=? function=?" part)

Reply:
Original:
```
 char *msg = "OpenTelemetry: pre hook invalid signature, class=%s  function=%s";
            log_invalid_message(msg, &params[2], &params[3]);
```

New:
```
php_error_docref(NULL, E_WARNING, "OpenTelemetry: pre hook invalid signature,  class=%s function=%s",
(Z_TYPE_P(&params[2]) == IS_NULL) ? "null" : Z_STRVAL_P(&params[2]), Z_STRVAL_P(&params[3]));
```

The diference is with `php_log_err` the error will be dumped only in php log file without to show anything in frontend, and with `php_error_docref` will dump error in php log file and print the warning also in frontend and also will show in what file and line the problem occur.

Reply:
I can do this:
```
<?php
OpenTelemetry\Instrumentation\hook(
    'notnull', // Need to be string
    'array_slice',
    pre: function(null $instance, array $params) {
        return [$params[0], $params[1], 1];
    },
    post: fn() => null
);

var_dump(array_slice([1,2,3], 1));
?>
```

With return:
```
array(2) {
  [0]=>
  int(2)
  [1]=>
  int(3)
}
```

Without any warning or error...